### ISSUE TITLE
[SUGGESTION] Emojis with custom formatting

### ISSUE THREAD

**Issue by Polda18** [oraxen/oraxen#1420]
### I have checked that...

- [X] ...such a feature does not exist already
- [X] ...such a feature request has not been submitted already

### Description

We are using custom chat formatting, where player's chat message is light gray (`&7`). When we tried to use the emojis first, it was darker than it should have been, because of our custom formatting. I don't even remember how we fixed it, I couldn't find it in the config files. However while the emojis are fixed, a text after the emojis is white instead of light gray. Can we make it so that the chat returns back to gray after the emoji? I couldn't figure out how to do it, and came to conclusion that Oraxen doesn't currently support it.

### Reason

Emojis are incompatible with custom chat formatting.

**Comment by Boy0000** [comment]
If ur on a Spigot server or Use a legacy chat handler / plugin no
If you use a Paper server and a modern chatplugin that is Paper only using their component logic yes
This is not somehting i plan on supporting as it is so much work keeping track of colorcodes and orders and whatnot

TLDR if u use a modern chatplugin and MODERN chat-handler in oraxens settings, this doesnt happen

**Comment by Polda18** [comment]
Well, we use CMI for custom chat, I believe it's a legacy formatting. Do we therefore need to enable legacy support?

**Comment by Boy0000** [comment]
LEGACY chathandler doesnt fix the issue here as legacy means chat is a string not fancy components
CMI is indeed not a paper chat plugin like say CarbonChat or ChatChat

LEGACY is linear aka "&cred:test:" and thus keeping track of every colorcode is unfeasable

**Comment by Polda18** [comment]
Looks like we'll have to do our own research. CMI doesn't support the new format, and plugins we found so far don't support the features we need. AdvancedChat does support the new format, and even integrates with Oraxen, but its chat swear filter doesn't support our needs for chat filtering, and the plugin doesn't support ChatSentry, which we do use for chat filtering. Looks like we'll have to keep looking. For the time being, we'll use CMI, sadly with the broken formatting until we find a better solution. Thank you.

**Comment by Boy0000** [comment]
ChatChat & CarbonChat are options aswell

**Comment by Polda18** [comment]
I wasn't able to find ChatChat at all, and CarbonChat (which is actually Carbon) doesn't really support our needs. Carbon seems to be still in beta for 1.20.4, and lacks any online documentation, making it difficult to set it up.

**Comment by Boy0000** [comment]
If need be I have my own chatplugin
Chatty which I have for another project but has a few dependencies
OraxenChat is also there but isnt updated in a while and still "unreleased betas" on our Discord
